src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -288,6 +290,74 @@ public void renameDirectory(String oldDirectoryPath, String targetDirectoryPath)
         FileUtils.moveDirectory(oldDirectory, targetDirectory);
     }
 
+    /**
+     * Look for sections that start and end with a section marker (e.g. %section-start% and %section-end%). Overrides the given file in filePath with a new file!
+     *
+     * @param filePath of file to look for replacable sections in.
+     * @param sections of structure String (section name) / Boolean (keep content in section or remove it).
+     */
+    public void replacePlaceholderSections(String filePath, Map<String, Boolean> sections) {
+        Map<Pattern, Boolean> patternBooleanMap = sections.entrySet().stream().collect(Collectors.toMap(e -> Pattern.compile(".*%" + e.getKey() + ".*%.*"), Map.Entry::getValue));
+        File file = new File(filePath);
+        File tempFile = new File(filePath + "_temp");
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+
+            Map.Entry<Pattern, Boolean> matchingStartPattern = null;
+            String line = reader.readLine();
+            while (line != null) {
+                // If there is no starting pattern matched atm, check if the current line is a start pattern.
+                if (matchingStartPattern == null) {
+                    for (Map.Entry<Pattern, Boolean> entry : patternBooleanMap.entrySet()) {
+                        if (entry.getKey().matcher(line).matches()) {
+                            matchingStartPattern = entry;
+                            break;
+                        }
+                    }
+                    // If a pattern is matched, don't write anything so that the section qualifier is removed.
+                    if (matchingStartPattern != null) {
+                        line = reader.readLine();
+                        continue;
+                    }
+                }
+                else {
+                    // If there is a starting pattern matched, check if an ending pattern is encountered.
+                    boolean endMatcherFound = false;
+                    for (Map.Entry<Pattern, Boolean> entry : patternBooleanMap.entrySet()) {
+                        if (entry.getKey().matcher(line).matches()) {
+                            endMatcherFound = true;
+                            break;
+                        }
+                    }
+                    if (endMatcherFound) {
+                        matchingStartPattern = null;
+                        line = reader.readLine();
+                        continue;
+                    }
+                }
+
+                if (matchingStartPattern == null || matchingStartPattern.getValue()) {
+                    writer.write(line);
+                    writer.newLine();
+                }
+
+                line = reader.readLine();
+            }
+
+            reader.close();
```

Comment:
move into final clause https://www.geeksforgeeks.org/flow-control-in-try-catch-finally-in-java/

Reply:
Good point!